### ISSUE TITLE
Submission: Group 23: Crime Forecast in Vancouver

### ISSUE THREAD

**Issue by SoloSynth1** [UBC-MDS/data-analysis-review-2023#16]
**Submitting authors: @phchen5, @SoloSynth1, @zywkloo, @MoNorouzi23**

**Repository:**
https://github.com/UBC-MDS/project-avalon
**Report link:**
https://ubc-mds.github.io/project-avalon/crime_forecasting.html
**Abstract/executive summary:**
This project aims to perform statistical analyses on the crimes committed in the City of Vancouver and develop a forecasting algorithm to forecast the number of crime given a set of lagged values.

We created a model to estimate the frequency of vehicle break-ins each month in Vancouver. It uses historical data to inform its predictions, applying methods that consider both recent occurrences and long-established trends. The chosen approach has an average deviation of about 27 incidents per month from the actual numbers, which is notable given the substantial month-to-month variation in the incident counts. The model's current accuracy demonstrates its potential, and we anticipate that incorporating additional factors, such as weather patterns or significant city events, could enhance its predictive capabilities.

**Editor:** @SoloSynth1 
**Reviewer:** @atabak-alishiri @sifanzzz

- [x] I agree to abide by [MDS's Code of Conduct](https://ubc-mds.github.io/resources_pages/code_of_conduct/) during the review process and in maintaining my package should it be accepted.

**Comment by atabak-alishiri** [comment]
## Data analysis review checklist

### Reviewer: @atabak-alishiri

### Conflict of interest

- [x] As the reviewer I confirm that I have no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [MDS code of conduct](https://ubc-mds.github.io/resources_pages/code_of_conduct/).

### General checks

- [x] **Repository:** Is the source code for this data analysis available? Is the repository well-organized and easy to navigate?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?

### Documentation

- [x] **Installation instructions:** Is there a clearly stated list of dependencies? 
- [x] **Example usage:** Do the authors include examples of how to use the software to reproduce the data analysis?
- [x] **Functionality documentation:** Is the core functionality of the data analysis software documented to a satisfactory level?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Code quality
- [x] **Readability:** Are scripts, functions, objects, etc., well named? Is it relatively easy to understand the code?
- [x] **Style guidelines:** Does the code adhere to well-known language style guides?
- [x] **Modularity:** Is the code suitably abstracted into scripts and functions?
- [x] **Tests:** Are there automated tests or manual steps described so that the function of the software can be verified? Are they of sufficient quality to ensure software robustness?

### Reproducibility
 
- [x] **Data:** Is the raw data archived somewhere? Is it accessible?
- [x] **Computational methods:** Is all the source code required for the data analysis available?
- [x] **Conditions:** Is there a record of the necessary conditions (software dependencies) needed to reproduce the analysis? Does there exist an easy way to obtain the computational environment needed to reproduce the analysis?
- [x] **Automation:** Can someone other than the authors easily reproduce the entire data analysis?

### Analysis report

- [x] **Authors:** Does the report include a list of authors with their affiliations?
- [x] **What is the question:** Do the authors clearly state the research question being asked?
- [x] **Importance:** Do the authors clearly state the importance of this research question?
- [x] **Background**: Do the authors provide sufficient background information so that readers can understand the report?
- [x] **Methods:** Do the authors clearly describe and justify the methodology used in the data analysis? Do the authors communicate any assumptions or limitations of their methodologies?
- [x] **Results:** Do the authors clearly communicate their findings through writing, tables and figures?
- [x] **Conclusions:** Are the conclusions presented by the authors correct? 
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?
- [x] **Writing quality:** Is the writing of good quality, concise, and engaging? 

### Estimated hours spent reviewing: 

### Review Comments: 

The "Crime Forecast in Vancouver" project demonstrates a thoughtful approach to crime forecasting using time-series models. However, there are several areas where the project could be improved:

- HTML file: An HTML file is present, but I was unable to recreate it, and there were no provided examples on how to generate one.
- Tables: The tables in the report are not displayed correctly.
- Introduction and Conclusion: The introduction effectively outlines the main goal of the project as "predicting theft from vehicles in Vancouver using historical data". The report successfully demonstrates the analytical and modeling process. However, in the conclusion (or introduction), adding a concise overview of the major findings related to crime prediction, rather than just the model's performance, would be beneficial. This addition would highlight the research's significance, rather than just the model's.
- Data Analysis Section: In the section where you address missing data, it would be useful to briefly describe how you manage these missing values, whether by removal or through some form of data imputation.

In conclusion, I gained valuable insights from reviewing your work and plan to apply them to my group project as well. Thank you very much for your excellent contribution!


#### Attribution

This was derived from the [JOSE review checklist](https://openjournals.readthedocs.io/en/jose/review_checklist.html) and the ROpenSci review checklist.

**Comment by sifanzzz** [comment]
## Data analysis review checklist

### Reviewer: <@sifanzzz> 

### Conflict of interest

- [x] As the reviewer I confirm that I have no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [MDS code of conduct](https://ubc-mds.github.io/resources_pages/code_of_conduct/).

### General checks

- [x] **Repository:** Is the source code for this data analysis available? Is the repository well organized and easy to navigate?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?

### Documentation

- [x] **Installation instructions:** Is there a clearly stated list of dependencies? 
- [x] **Example usage:** Do the authors include examples of how to use the software to reproduce the data analysis?
- [x] **Functionality documentation:** Is the core functionality of the data analysis software documented to a satisfactory level?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Code quality
- [x] **Readability:** Are scripts, functions, objects, etc., well named? Is it relatively easy to understand the code?
- [x] **Style guidelides:** Does the code adhere to well known language style guides?
- [x] **Modularity:** Is the code suitably abstracted into scripts and functions?
- [x] **Tests:** Are there automated tests or manual steps described so that the function of the software can be verified? Are they of sufficient quality to ensure software robsutness?

### Reproducibility
 
- [x] **Data:** Is the raw data archived somewhere? Is it accessible?
- [x] **Computational methods:** Is all the source code required for the data analysis available?
- [x] **Conditions:** Is there a record of the necessary conditions (software dependencies) needed to reproduce the analysis? Does there exist an easy way to obtain the computational environment needed to reproduce the analysis?
- [x] **Automation:** Can someone other than the authors easily reproduce the entire data analysis?

### Analysis report

- [x] **Authors:** Does the report include a list of authors with their affiliations?
- [x] **What is the question:** Do the authors clearly state the research question being asked?
- [x] **Importance:** Do the authors clearly state the importance for this research question?
- [x] **Background**: Do the authors provide sufficient background information so that readers can understand the report?
- [x] **Methods:** Do the authors clearly describe and justify the methodology used in the data analysis? Do the authors communicate any assumptions or limitations of their methodologies?
- [x] **Results:** Do the authors clearly communicate their findings through writing, tables and figures?
- [x] **Conclusions:** Are the conclusions presented by the authors correct? 
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?
- [x] **Writing quality:** Is the writing of good quality, concise, engaging? 

### Estimated hours spent reviewing: 1.5hour

### Review Comments: 
Your report is well-structured, thorough, and provides a comprehensive overview of your crime forecasting project in Vancouver. Great work overall! 

Here are some suggestions to enhance clarity and further strengthen your report:

Introduction and conclusion: It is great that in the introduction part, you clearly state that "the primary objective of this project is to forecast instances of theft from vehicles in Vancouver by analyzing historical data", and the report shows clearly the process of analysis and modelling. But in the conclusion(or in the introduction), it would be better to consider adding a brief statement summarizing the key findings regarding the crime forecasting results, not just a conclusion of the performance of models. It could help to emphasize the importance of this research(not the importance of this model).

README file: it would be nice to also show the author names here as well as the references in the README file.

Data Analysis Section: When discussing missing values in the dataset, could be helpful to briefly mention your approach to handling these missing values. Are they dropped, or is there an imputation strategy in place?

Tables in report: the report would be easier to read and look better displayed if tables could be well rendered.

#### Attribution

This was derived from the [JOSE review checklist](https://openjournals.readthedocs.io/en/jose/review_checklist.html) and the ROpenSci review checklist.
